src/auth/project-config.ts

```diff
@@ -0,0 +1,131 @@
+/*!
+ * Copyright 2022 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as validator from '../utils/validator';
+import { AuthClientErrorCode, FirebaseAuthError } from '../utils/error';
+import {
+  SmsRegionsAuthConfig,
+  SmsRegionConfig,
+} from './auth-config';
+import { deepCopy } from '../utils/deep-copy';
+
+/**
+ * Interface representing the properties to update on the provided project config.
+ */
+export interface UpdateProjectConfigRequest {
+  /**
+   * The SMS configuration to update on the project.
+   */
+  smsRegionConfig?: SmsRegionConfig;
+}
+
+/**
+ * Response received from get/update project config.
+ * We are only exposing the SMS Region config for now.
+ */
+export interface ProjectConfigServerResponse {
+  smsRegionConfig?: SmsRegionConfig;
+}
+
+/**
+ * Request sent to update project config.
+ * We are only updating the SMS Regions config for now.
+ */
+export interface ProjectConfigClientRequest {
+  smsRegionConfig?: SmsRegionConfig;
+}
+
+/**
+* Represents a project configuration.
+*/
+export class ProjectConfig {
+  /**
+   * The SMS Regions Config to update on the project config.
+   * Configures the regions where users are allowed to send verification SMS.
+   * This is based on the calling code of the destination phone number.
+   */
+  public readonly smsRegionConfig?: SmsRegionConfig;
+
+  /**
+   * Validates a project config options object. Throws an error on failure.
+   *
+   * @param request - The project config options object to validate.
+   */
+  private static validate(request: any): void {
+    if (!validator.isNonNullObject(request)) {
+      throw new FirebaseAuthError(
+        AuthClientErrorCode.INVALID_ARGUMENT,
+        '"UpdateProjectConfigRequest" must be a valid non-null object.',
```

Comment:
This is the check on the whole updateProjectConfigRequest object.

Reply:
If we are only validating UpdateProjectConfigRequest, then should the type of the request parameter to validate be UpdateProjectConfigRequest rather than "any"?

@renkelvin @lahirumaramba any reason to leave the type as any?

Reply:
Good catch! I think the type can be set to: `UpdateProjectConfigRequest`

Reply:
Thank you Lahiru and Pavithra!  Updated to `UpdateProjectConfigRequest`.